From 483c571a6cfadb9e6d24f472d13bb46072ab650a Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:00:20 -0400
Subject: [ethernet] mlx4: Deregister multicast vxlan steering rules when going down

Message-id: <4ed2d68fb0c39331ad4e064eac7c314cdb5badcb.1410425016.git.avadai@redhat.com>
Patchwork-id: 93717
O-Subject: [Patch RHEL7.1 V1 084/197] net/mlx4_en: Deregister multicast vxlan steering rules when going down
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Or Gerlitz <ogerlitz@mellanox.com>

When mlx4_en_stop_port() is called, we need to deregister also the
tunnel steering rules that relate to multicast.

Signed-off-by: Or Gerlitz <ogerlitz@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit de123268300fd33b7f7668fda3264059daffa6ef)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_netdev.c b/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
index b4f6547..fa5ee71 100644
--- a/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
@@ -1788,6 +1788,8 @@ void mlx4_en_stop_port(struct net_device *dev, int detach)
   mc_list[5] = priv->port;
   mlx4_multicast_detach(mdev->dev, &priv->rss_map.indir_qp,
           mc_list, MLX4_PROT_ETH, mclist->reg_id);
+  if (mclist->tunnel_reg_id)
+   mlx4_flow_detach(mdev->dev, mclist->tunnel_reg_id);
  }
  mlx4_en_clear_list(dev);
  list_for_each_entry_safe(mclist, tmp, &priv->curr_list, list) {
-- 
1.7.1